core/arch/arm/plat-sunxi/main.c

```diff
@@ -190,41 +163,8 @@ static void main_tee_entry(struct thread_smc_args *args)
 	tee_entry(args);
 }
 
-
-/* Override weak function in tee/entry.c */
+/* main_tee_entry() supports 3 platform-specific functions */
 void tee_entry_get_api_call_count(struct thread_smc_args *args)
 {
-	args->a0 = tee_entry_generic_get_api_call_count() + 2;
-}
-
-/* Override weak function in tee/entry.c */
-void tee_entry_get_api_uuid(struct thread_smc_args *args)
-{
-	args->a0 = TEESMC_OPTEE_UID_R0;
-	args->a1 = TEESMC_OPTEE_UID_R1;
-	args->a2 = TEESMC_OPTEE_UID_R2;
-	args->a3 = TEESMC_OPTEE_UID32_R3;
-}
-
-/* Override weak function in tee/entry.c */
-void tee_entry_get_api_revision(struct thread_smc_args *args)
-{
-	args->a0 = TEESMC_OPTEE_REVISION_MAJOR;
-	args->a1 = TEESMC_OPTEE_REVISION_MINOR;
-}
-
-/* Override weak function in tee/entry.c */
-void tee_entry_get_os_uuid(struct thread_smc_args *args)
-{
-	args->a0 = TEESMC_OS_OPTEE_UUID_R0;
-	args->a1 = TEESMC_OS_OPTEE_UUID_R1;
-	args->a2 = TEESMC_OS_OPTEE_UUID_R2;
-	args->a3 = TEESMC_OS_OPTEE_UUID_R3;
-}
-
-/* Override weak function in tee/entry.c */
-void tee_entry_get_os_revision(struct thread_smc_args *args)
-{
-	args->a0 = TEESMC_OS_OPTEE_REVISION_MAJOR;
-	args->a1 = TEESMC_OS_OPTEE_REVISION_MINOR;
+	args->a0 = tee_entry_generic_get_api_call_count() + 3;
```

Comment:
There are 3 specific `TEESMC32_*` values handled in https://github.com/jforissier/optee_os/blob/refactoring/core/arch/arm/plat-sunxi/main.c#L136 hence +3. Did I miss something?